### ISSUE TITLE
Requesting assistance: issue with missing journals/books in article/book chapter citations

### ISSUE THREAD

**Issue by JahnSLSP** [citation-style-language/styles#6745]
Hello, 
We would like to request assistance with an issue we see with two citation styles, where our citations are lacking the journal/book for article/book chapter citations. 
The affected styles are modern-language-association-7th-edition.csl and harvard-kings-college-london.csl. 
We are working on an Alma/Primo VE environment (Ex Libris) and had previously contacted Ex Libris about this issue. They were able to point us to a solution for our other citation styles (APA 7th Edition, MLA 8th Edition and Chicago) but insist that since those are working, the issue for the remaining two styles must be with the styles themselves. As we are not entirely sure how Ex Libris implements the citation styles, we are sadly not able to offer much more insight into the issue. 
For a set of examples see: 
https://slsp-network-psb.primo.exlibrisgroup.com/permalink/41SLSP_NETWORK/1ufb5t2/alma991149337319705501
https://slsp-network-psb.primo.exlibrisgroup.com/permalink/41SLSP_NETWORK/1ufb5t2/alma991131396419705501
https://slsp-network-psb.primo.exlibrisgroup.com/permalink/41SLSP_NETWORK/1ufb5t2/alma991075223479705501

![image](https://github.com/citation-style-language/styles/assets/148247698/82d0268e-14ad-434d-ac87-fce8bd3b1b7f)

Thank you & Kind Regards

**Comment by denismaier** [comment]
It's hard to tell without knowing more details. MLA7 should contain journal titles and book titles for articles and chapters. How does the data look like? Which csl processor is used by Ex Libris?

**Comment by JahnSLSP** [comment]
We have now been informed that Ex Libris uses citeproc: https://github.com/michel-kraemer/citeproc-java.

**Comment by denismaier** [comment]
I might be wrong, but I don't think citeproc-java works with MARC21 directly, at least I couldn't find that on the project page. It seems to work with RIS, Endnote and BibTeX files. It would be interesting to know how they pass the data to citeproc-java.

**Comment by adam3smith** [comment]
Yes, the way the Primo CSL integration works is that they send the PNX data (not the MARC) together with the citation style to the ExLibris server and then return the citation string. How they transform the PNX into the data format taken by citeproc-java (which is CSL-JSON) is a black box for us.

In this case, though, I can reverse-engineer what's going on: The articles are treated as `book` item types in the citation styles. Depending on how the styles are coded, you still get the publication title (`continer-title` in CSL), in others you don't, because a book doesn't have a publication title, conceptually. You'll note that the APA style, e.g., while printing the publication title, is also incorrect: the shouldn't be an "In" in front of journal or newspaper titles -- that's for chapters in books and similar publication types.

So what you need to figure out together with ExLibris is how you can get them to treat these as CSL `article-journal` or `article-newspaper` depending on what they are.  Since these are labelled correctly as article in the PNX, that should be doable.

One clue might be that Zotero is also not recognizing this properly because you're using the unusual `type: articles` rather than the expected `type: article` for the item type, so if you can fix that in your mapping that might already fix this.

**Comment by JahnSLSP** [comment]
Dear Denis, Dear Sebastian, 

Thank you very much for your input! Just a quick follow up: The ressource type you see is _articles_ and not _ARTICLES_, correct? This gives us grounds to rediscuss the issue with Ex Libris. We'd come back to you in case we'd need further assistance.
